src/scripting/ScriptingProvider.js

```diff
@@ -19,17 +19,42 @@ module.exports = class ScriptingProvider {
     this.utterances = {}
     this.matchFn = null
     this.asserters = {}
+    this.globalAsserter = {}
 
     this.scriptingEvents = {
-      assertConvoBegin: (convo) => {
-        return Promise.resolve()
+      assertConvoBegin: ({convo, container}) => {
+        const convoAsserter = convo.beginAsserter
+          .filter(a => this.asserters[a.name].assertConvoBegin)
+          .map(a => this.asserters[a.name].assertConvoBegin({convo, container, args: a.args}))
+
```

Comment:
> what happens if the asserter doesn't return a promise ? or if it throws an exception ? does this work with this implementation ?

currently it will just work when asserter returns promise. I read this: https://gist.github.com/MarkoCen/ec27b8cd42855fde8a245d43b7b081d0. Is there any smart way to wrap a sync function into a promise and catch errors through promise.reject?

Reply:
> what happens if the asserter doesn't return a promise ? or if it throws an exception ? does this work with this implementation ?

I tested it with a asserter which is not resp with a Promise and error case and success case are still  working surprisely. So to your previous questin -> YES

Reply:
Yes, I was confident that in the way your code is written it will still work - promises work this way out of the box - but now we know for sure.